creating_functions_for_EDA.md

```diff
@@ -0,0 +1,223 @@
+inspection on those columns with excessive missing values. With specification of the
+threshold of the missing value percentage, the following
+function will give you a list of columns that have missing values over that
+threshold:
+
+
+```
+def view_columns_w_many_nans(df, missing_percent):
+    '''
+    Checks which columns have over specified percentage of missing values
+    Takes df, missing percentage
+    Returns columns as a list
+    '''
+    mask_percent = df.isnull().mean()
+    series = mask_percent[mask_percent > missing_percent]
+    columns = series.index.to_list()
+    print(columns) 
+    return columns
+```
+There are many ways to deal with missing values. If you decide to drop the
+columns with too many missing values (over certain threshold of which you specify), this function can help to achieve this
+task:
+```
+def drop_columns_w_many_nans(df, missing_percent):
+    '''
+    Takes df, missing percentage(default=.9)
+    Drops the columns whose missing value is bigger than missing percentage
+    Returns df
+    '''
+    series = view_columns_w_many_nans(df, missing_percent=missing_percent)
+    list_of_cols = series.index.to_list()
+    df.drop(columns=list_of_cols)
+    print(list_of_cols)
+    return df
+```
+However, there are many downsides of deleting missing values from your dataset,
+such as reduced statistical power. If you decide to impute the missing value
+instead, check out Sklearn's
+[`SimpleImputer`](https://scikit-learn.org/stable/modules/generated/sklearn.impute.SimpleImputer.html)
+module,
+which is an easy-to-use tool to impute missing values whichever way strikes
+yours fancy. 
+
+
+In addition, if you'd like to read more about how to deal with missing values, check out this
+helpful [slide](https://liberalarts.utexas.edu/prc/_files/cs/Missing-Data.pdf)
+created by Melissa Humphries at Population Research Center. 
+
+
+---
+## Functions for Data Visualization
+
+Human brain is very good at identify patterns, and this is why visualizing your
+dataset during the EDA process and identify the patterns can be very beneficial. For instance, histogram
+makes analyzing the distribution of the data an easier task; Boxplot is great for
+identify outliers; Scatter plot is very useful when it comes to checking the
+correlations between two variables. [Matplotlib](https://matplotlib.org/) and [Seaborn](https://seaborn.pydata.org/) are your best friends
+when it comes to data visualization. However, creating individual plot(s) for
+each features can become tedious if you have a large number of features. In this
+section, I will walk you through a few functions for creating group plots that can help you to
+kill many birds with one function. 
+
+We often want to look at the distributions of columns with numerical values.
+The following function will create a group plot for all numerical columns in your
+dataset. This function is adapted from Dominik Gawlik's [blog
```

Comment:
It looks like this is a parenthetical remark, so suggesting it be wrapped as such

```suggestion
dataset. (This function is adapted from Dominik Gawlik's [blog
```